[Performance] MiqUserRole.seed speedup

kbrock opened the pull request:
- remove feature references from user role yaml file that do not exist
- remove N+1 lookup of feature
- add `changed?` to save
- did not do: remove roles N+1 (23% more rows coming back)

|     ms | bytes | objects |query | query ms |     rows |`comments`
|    ---:|   ---:|   ---:|  ---:|   ---:|      ---:| ---
|  446.8 | 11,737,352* | 130,318 |  102 | 220.3 |    1,845 |before
|  361.1 |  9,677,469* | 105,722 |   57 | 166.6 |    1,841 |after
| 19% | 18% | 19% | 44% | 24% | 0% | delta

speeding up seeds. but this is 100ms, not sure if it is worth the risk
related to #15586 and https://bugzilla.redhat.com/show_bug.cgi?id=1422671

chrisarcand commented:
Do we even need `seed_from_array`? Seems like it's just a helper method used in this class and can be combined. `merge_features` always true? Is this method a useful thing that people use in the console...?

bdunne commented on app/models/miq_user_role.rb:
```diff
@@ -75,26 +75,29 @@ def self.with_allowed_roles_for(user_or_group)
   def self.seed
-    seed_from_array(YAML.load_file(FIXTURE_YAML))
+    roles = all.index_by(&:name)
+    seed_from_array(roles, YAML.load_file(FIXTURE_YAML))
```
I know it's not related to your change, but it's bothering me that we have duplicate code here.  Do you think it would make sense to move `FIXTURE_YAML` into the `FIXTURE_PATH`?

bdunne commented on db/fixtures/miq_user_roles.yml:
```diff
@@ -14,7 +14,6 @@
   - embedded_automation_manager
   - availability_zone
   - host_aggregate
-  - compute
```
I don't really know why this file changed.  @kbrock can you explain?

kbrock replied:
of note, the opposite shows that 1,051 features are identified in `miq_product_features.yml` that are not referenced in `miq_user_roles.yml` - which makes sense. just wanted to share for the record

```bash
for i in `grep ':identifier:' db/fixtures/miq_product_features.yml | awk '{print $2}' | sort -u` ; do
  grep -q -- "- ${i}$" db/fixtures/miq_user_roles.yml || echo "NOT REFERENCED: ${i}"
done
```

kbrock commented on db/fixtures/miq_user_roles.yml:
```diff
@@ -14,7 +14,6 @@
   - embedded_automation_manager
   - availability_zone
   - host_aggregate
-  - compute
```
Yea that seems confusing to me too.

There is no `compute` feature in `db/fixtures/miq_product_features.yml`.
So their reference in `miq_user_roles.yml` will just be a waisted lookup.

I used the following to determine if any invalid:

```
for i in `grep '^  -' db/fixtures/miq_user_roles.yml | sort -u`  ; do
  grep -q -- "${i}$" db/fixtures/miq_user_roles.yml || echo "MISSING: ${i}"
done
```

kbrock commented:
ran through both scripts to make sure there were no surprises.

kbrock commented:
fixed conflict.

verified that we are not removing any user roles that are existing features.

@bdunne Would you prefer that I split apart the data changes and model changes?
